SmartcarAuth/OEM/OEM.swift

```diff
@@ -78,3 +78,17 @@ public class OEM {
     }
     
 }
+
+extension Array where Element == OEMName {
+    /**
+     Return sorted list of contained `OEMName`s by `displayName`, while keeping `mock` at the end
+     */
+    public func sortedByDisplayName() -> [Element] {
```

Comment:
@sankethkatta Should we just hand-sort the OEMs in the array rather than having this function?

Reply:
But the OEMs look like they are in a dictionary/object, right? would that structure provide ordering? (I'm not super familiar with Swift/iOS data types!)

In which case we should have the function to sort the values.

Reply:
@zz60 hand-sorting was not the goal here. This is to sort the list of OEMs used by the app when using the picker UI. It currently uses the list of OEMs in the order provided by the app, though sorting them seems to make more sense.

@sankethkatta No it would not, there are no ordered hashmap in Swift. but that leads us to a question I have for a while: why is there a need for OEM + OEMName + OEMConfig ? a single enum in Swift can actually combine all this.